### ISSUE TITLE
Event listeners are called back with this as *ElementImpl, not *Element.

### ISSUE THREAD

**Issue by mbostock** [tmpvar/jsdom#1369]
Grats on the new 8.0 major release. I’d like to upgrade, but I noticed this bug. Here’s a simple test case:

```js
var document = require("jsdom").jsdom();

console.log(document.body.constructor.name); // "HTMLBodyElement";

document.body.addEventListener("click", function() {
  console.log(this.constructor.name, this === document.body); // "HTMLBodyElementImpl", false
});

document.body.dispatchEvent(new document.defaultView.CustomEvent("click"));
```

In addition to failing the equality test, it also prevents you from doing things like accessing raw properties assigned to elements, such as `element.__data__` in D3.

**Comment by domenic** [comment]
Thanks for finding!! @Sebmaster, our impl leaked through; plz fix asap.

**Comment by mbostock** [comment]
Wild guess but could be as simple as replacing this:

```js
listener.callback.call(eventImpl.currentTarget, idlUtils.wrapperForImpl(eventImpl));
```

With this:

```js
listener.callback.call(idlUtils.wrapperForImpl(eventImpl.currentTarget), idlUtils.wrapperForImpl(eventImpl));
```

**Comment by domenic** [comment]
Yeah seems very likely. We do need a test; I'm very surprised there are no tests that `x.aEL(f)` calls `f` with `this === x`.

**Comment by domenic** [comment]
I guess I can fix this during the next boring TC39 presentation today.

**Comment by domenic** [comment]
Fixed in 8.0.1! Thanks so much for bravely jumping on the new release to give us this kind of testing :)

**Comment by mbostock** [comment]
:+1: Works great. Thank you for the fast fix!
